File: advanced-programming-methods/labs/toy-interpreter-sharp/model/expression/Operation.cs
using System;

namespace Model
{
    /// <summary>
    /// The type Operation.
    /// </summary>
    public static class Operation
    {
        /// <summary>
        /// The constant ADD.
        /// </summary>
        public const int ADD = 1;

        /// <summary>
        /// The constant SUBTRACT.
        /// </summary>
        public const int SUBTRACT = 2;

        /// <summary>
        /// The constant MULTIPLY.
        /// </summary>
        public const int MULTIPLY = 3;

        /// <summary>
        /// The constant DIVIDE.
        /// </summary>
        public const int DIVIDE = 4;

        /// <summary>
        /// Operation to string.
        /// </summary>
        /// <param name="op"> the op </param>
        /// <returns> the string </returns>
        public static string OperationToString(int op)
        {
            switch (op)
            {
                case ADD:
                    return "+";
                case SUBTRACT:
                    return "-";
                case MULTIPLY:
                    return "*";
                case DIVIDE:
                    return "/";
                default:
                    return "";
            }
        }
    }

}

File: advanced-programming-methods/labs/toy-interpreter-java/src/model/expression/Operation.java
package model.expression;

/**
 * The type Operation.
 */
public class Operation
{
    /**
     * The constant ADD.
     */
    public static final int ADD      = 1;
    /**
     * The constant SUBTRACT.
     */
    public static final int SUBTRACT = 2;
    /**
     * The constant MULTIPLY.
     */
    public static final int MULTIPLY = 3;
    /**
     * The constant DIVIDE.
     */
    public static final int DIVIDE   = 4;

    /**
     * Operation to string.
     *
     * @param op the op
     * @return the string
     */
    public static String operationToString(int op)
    {
        switch (op)
        {
            case ADD:
                return "+";
            case SUBTRACT:
                return "-";
            case MULTIPLY:
                return "*";
            case DIVIDE:
                return "/";
            default:
                return "";
        }
    }
}